Import the Pandas library to provide the data frame to work in the csv file as 
it will read the file as a dataframe to retrieve the records

In [1]:
import pandas as pd
import numpy as np

Reading the csv file

In [2]:
df = pd.read_csv("sales.csv")

Information about the data frame 

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   year         12 non-null     int64 
 1   month        12 non-null     object
 2   sales        12 non-null     int64 
 3   expenditure  12 non-null     int64 
dtypes: int64(3), object(1)
memory usage: 512.0+ bytes


check the dataframe first 5 entries 

In [4]:
df.head(5)

year month  sales  expenditure
0  2018   jan   6226         3808
1  2018   feb   1521         3373
2  2018   mar   1842         3965
3  2018   apr   2051         1098
4  2018   may   1728         3046

In [ ]:
Checks for how many unique years are mentioned here

In [5]:
df['year'].unique()

array([2018])

Finding the record when the max sales as taken place

In [6]:
df['sales'].max()

7479

In [11]:
df.loc[df['sales'].idxmax()]

year           2018
month           jul
sales          7479
expenditure    2084
Name: 6, dtype: object

Find the record for the macimum expenditure

In [12]:
df.loc[df['expenditure'].idxmax()]

year           2018
month           mar
sales          1842
expenditure    3965
Name: 2, dtype: object

Converting the sales column into a list

In [3]:
sale_list = df['sales'].tolist()

In [4]:
print(sale_list)

[6226, 1521, 1842, 2051, 1728, 2138, 7479, 4434, 3615, 5472, 7224, 1812]


Output the total sales across all months

In [5]:
total_sales = df['sales'].sum()

In [6]:
total_sales

45542

Monthly changes as a percentage - Creating a new column for percentages

In [15]:
df['sales_pct']= df['sales'].pct_change().round(2)*100

In [16]:
print(df)

    year month  sales  expenditure  sales_pct
0   2018   jan   6226         3808        NaN
1   2018   feb   1521         3373      -76.0
2   2018   mar   1842         3965       21.0
3   2018   apr   2051         1098       11.0
4   2018   may   1728         3046      -16.0
5   2018   jun   2138         2258       24.0
6   2018   jul   7479         2084      250.0
7   2018   aug   4434         2799      -41.0
8   2018   sep   3615         1649      -18.0
9   2018   oct   5472         1116       51.0
10  2018   nov   7224         1431       32.0
11  2018   dec   1812         3532      -75.0


The average sales and expenditure

In [17]:
df.describe()

year        sales  expenditure   sales_pct
count    12.0    12.000000    12.000000   11.000000
mean   2018.0  3795.166667  2513.250000   14.818182
std       0.0  2285.725579  1046.608994   88.605664
min    2018.0  1521.000000  1098.000000  -76.000000
25%    2018.0  1834.500000  1594.500000  -29.500000
50%    2018.0  2876.500000  2528.500000   11.000000
75%    2018.0  5660.500000  3412.750000   28.000000
max    2018.0  7479.000000  3965.000000  250.000000

calculating the average sales and expenditure

In [21]:
avg_sales = df['sales'].mean()

In [22]:
avg_sales

3795.1666666666665

avg_expenditure is 2513.25

Months with the highest and lowest sales

In [23]:
max_sales_month = df.loc[df['sales'].idxmax()]

In [24]:
max_sales_month

year            2018
month            jul
sales           7479
expenditure     2084
sales_pct      250.0
Name: 6, dtype: object

In [26]:
min_sales_month = df.loc[df['sales'].idxmin()]

In [27]:
min_sales_month

year           2018
month           feb
sales          1521
expenditure    3373
sales_pct     -76.0
Name: 1, dtype: object